File: node-serverless-scheduler/db/reports/models/ReportHistory.ts
import {
  DataTypes,
  InferAttributes,
  InferCreationAttributes,
  Model,
} from 'sequelize';

export class ReportHistory extends Model<
  InferCreationAttributes<ReportHistory>,
  InferAttributes<ReportHistory>
> {
  declare id: string;
  declare reportName: string;
  declare userName: string;
  declare filePath: string;
  declare status: string;
  declare dateFrom: Date;
  declare dateTo: Date;
  declare createdAt: Date;
  declare updatedAt: Date;
}

export const initModel = (sequelize) => {
  ReportHistory.init(
    {
      id: {
        type: DataTypes.UUID,
        primaryKey: true,
      },
      reportName: DataTypes.STRING,
      userName: DataTypes.STRING,
      filePath: DataTypes.STRING(2048),
      status: DataTypes.STRING,
      dateFrom: DataTypes.DATEONLY,
      dateTo: DataTypes.DATEONLY,
      createdAt: DataTypes.DATE,
      updatedAt: DataTypes.DATE,
    },
    {
      sequelize,
      tableName: 'report_history',
      underscored: true,
    }
  );
};

File: backoffice-srv/src/Pi.BackofficeService/Pi.BackofficeService.Application/Models/ReportHistory.cs
namespace Pi.BackofficeService.Application.Models;

public record ReportHistory
{
    public required Guid Id { get; init; }
    public required ReportStatus Status { get; init; }
    public required string? Name { get; init; }
    public required ReportType? Type { get; init; }
    public required string? UserName { get; init; }
    public required DateOnly? DateFrom { get; init; }
    public required DateOnly? DateTo { get; init; }
    public required DateTime? GeneratedAt { get; init; }
    public required DateTime CreatedAt { get; init; }
    public required DateTime? UpdatedAt { get; init; }
}
